### PR TITLE
Pass get_type() functions, if any, to glib_wrapper!() for shared/boxe…

### PR DESCRIPTION
**Pull request by sdroege**
…d types

See https://github.com/gtk-rs/glib/pull/199

### CODE REVIEW

File: src/codegen/general.rs [hunk 1]
```diff
@@ -143,6 +148,9 @@ pub fn define_shared_type(
     try!(writeln!(w, "\tmatch fn {{"));
     try!(writeln!(w, "\t\tref => |ptr| ffi::{}(ptr),", ref_fn));
     try!(writeln!(w, "\t\tunref => |ptr| ffi::{}(ptr),", unref_fn));
+    if let &Some(ref get_type_fn) = get_type_fn {
```
Comment (EPashkin) [inline src/codegen/general.rs]: Maybe better `Some() = *get_type_fn()`
And in `codegen\translate_from_glib.rs:34:25 if let &Some(_) = array_length` please too .

**Comment by EPashkin** [comment]
Other that minor nit all looks good and only rebase needed.

File: src/codegen/general.rs (hunk 1 above)
Comment (sdroege) [inline src/codegen/general.rs]: Does it make any actual difference, or is it just style?

Comment (EPashkin) [inline src/codegen/general.rs]: Just `clippy` warning

**Comment by sdroege** [comment]
Should also be good now

**Comment by EPashkin** [comment]
Thanks, @sdroege

**Pull request merged**